### PR TITLE
Fix: Don't set the target field if the existing target document is false

### PR DESCRIPTION
**Pull request by 2002Bishwajeet**
If you create a user, set the target then delete the user and the target from console and then create a user again, you get a type error in the User Model
image

This happens cause it tries to populate the target field but the response was false

Adds a check if the document exists then set the target

## Test Plan

WIP

## Related PRs and Issues

https://discord.com/channels/564160730845151244/1062789941727338577/1247967828015972362

## Checklist

- [x] Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?
- [ ] If the PR includes a change to an API's metadata (desc, label, params, etc.), does it also include updated API specs and example docs?

### CODE REVIEW

File: tests/e2e/Services/Account/AccountCustomClientTest.php [hunk 1]
```diff
@@ -327,7 +327,6 @@ public function testGetAccountLogs($data): array
         $this->assertEquals('desktop', $response['body']['logs'][2]['deviceName']);
         $this->assertEquals('', $response['body']['logs'][2]['deviceBrand']);
         $this->assertEquals('', $response['body']['logs'][2]['deviceModel']);
-        $this->assertEquals($response['body']['logs'][2]['ip'], filter_var($response['body']['logs'][2]['ip'], FILTER_VALIDATE_IP));
```
Comment (stnguyen90) [inline tests/e2e/Services/Account/AccountCustomClientTest.php:330]: How come this was removed?

Comment (2002Bishwajeet) [reply tests/e2e/Services/Account/AccountCustomClientTest.php:330]: We already do the assertions check [here](https://github.com/appwrite/appwrite/pull/8236/files/485b8eeb8b2461f46a877cbc017baeaf487a9c33#diff-434428e8237205f1c4c6540329c60816d1e243383b6a7b044b844a594f4a35edL296) . Or its different? 👀

**Comment by 2002Bishwajeet** [comment]
I will add the unit tests for this PR by today. Had some busy days🙏

File: app/controllers/api/users.php [hunk 2]
```diff
@@ -138,7 +138,9 @@ function createUser(string $hash, mixed $hashOptions, string $userId, ?string $e
                 $existingTarget = $dbForProject->findOne('targets', [
                     Query::equal('identifier', [$email]),
                 ]);
-                $user->setAttribute('targets', [...$user->getAttribute('targets', []), $existingTarget]);
+                if($existingTarget) {
+                    $user->setAttribute('targets', [...$user->getAttribute('targets', []), $existingTarget]);
```
Comment (stnguyen90) [inline app/controllers/api/users.php:142]: `setAttribute()` has support for appending. See the [function signature](https://github.com/utopia-php/database/blob/45def2f7c6bc5f631dbb67e5df0e8e7331af5f63/src/Database/Document.php#L215). and example:

https://github.com/appwrite/appwrite/blob/0376189dbcb791d6ea28fd69332d7df1b208f1c3/app/init.php#L1075-L1080

**Review by stnguyen90 — CHANGES_REQUESTED** [review]

File: app/controllers/api/users.php (hunk 2 above)
Comment (2002Bishwajeet) [reply app/controllers/api/users.php]: Done

**Comment by 2002Bishwajeet** [comment]
I don't think a unit test would be needed (needs too much repetitive code create user, target, delete user, target then create user to check again).

**Review by stnguyen90 — APPROVED** [review]
Theoretically, the existing tests should confirm everything still works as intended.

**Pull request merged**